generators/base-application/types.d.ts

```diff
@@ -1,5 +1,6 @@
 /* eslint-disable @typescript-eslint/consistent-type-imports */
 import type { ExportApplicationPropertiesFromCommand } from '../../lib/command/types.js';
+import type { ApplicationOptions } from '../../lib/types/application/options.js';
```

Comment:
Cycle

Reply:
JHipsterGeneratorOptions which was removed extends ApplicationOptions, if we remove ApplicationOptions from types too it will cause 10 compilation issues.
We need to modularize before removing.

Reply:
The root cause of my `any`on the 2 previous PR

Reply:
It’s MUCH better to keep types consistent and postpone refactoring than break types.